### PR TITLE
Extra features

### PR DESCRIPTION
**Pull request by rdehuyss**
The new PR:
- The existing features including fixes using your feedback
- I also added an option to filter on to email address in the UI (it may bounce as I don't use an `AbortController` but I don't think debouncing is necessary)
- I added an authentication option
- I made sure the page is not indexed by Google
- I updated the dependencies

### CODE REVIEW

File: src/ui/mails/Mails.js [hunk 1]
```diff
@@ -21,8 +24,17 @@ class Mails extends React.Component {
...
     }
 
-    _fetchMails() {
-        fetch('/api/mails')
+    filterOnEmailTo(event) {
+        if(event.target.value.length > 3) {
+            this._fetchMails(`to=%${event.target.value}%`);
+        } else if (event.target.value.length < 1) {
+            this._fetchMails();
+        }
```
Comment (janjaali) [inline src/ui/mails/Mails.js:32]: This function is not defined for string lengths of 1, 2 and 3. Could you please make this function exhaustive?

File: src/ui/mails/Mails.js [hunk 2]
```diff
@@ -47,65 +59,86 @@ class Mails extends React.Component {
...
                                     .map(value => value.to)
                                     .map((tos, index) => (
-                                        <ul key={index}>
-                                            {tos.map((to, index) =>
-                                                (<li key={index}>{to.email}</li>))}
-                                        </ul>)
+                                        <span key={index}>
+                                            {tos.map((to, index) => 
+                                                (<span key={index}>{to.email}</span> ))}
+                                        </span>)
```
Comment (janjaali) [inline src/ui/mails/Mails.js:128]: Multiple email addresses should be visualized as list, the use of `span` does not match this requirement.

**Review by janjaali — CHANGES_REQUESTED** [review]
Thanks for the new PR, I just have left some open comments regarding the latest additions ;) Cheers :)

File: src/api/app.js [hunk 3]
```diff
@@ -29,20 +34,122 @@ app.post('/v3/mail/send', (req, res) => {
...
+if(process.env.AUTHENTICATION) {
+    const basicAuth = require('express-basic-auth')
+    const authenticationParams = process.env.AUTHENTICATION.split(";");
+    const users = {};
+    for (const auth of authenticationParams) {
+        const userPasswordCombo = auth.split(":");
+        users[userPasswordCombo[0]] = userPasswordCombo[1];
+    }
+    app.use(basicAuth({challenge: true, users}));
+}
```
[the remark this answers is not in the document]
Comment (rdehuyss) [reply src/api/app.js:46]: Well, it was there before but we were asked to also protect the API (except of course the SendGrid part). Once logged in the UI, the basic authentication is also valid for the api.

What do you think?

File: src/ui/mails/Mails.js (hunk 1 above)
Comment (rdehuyss) [reply src/ui/mails/Mails.js:32]: Will do it tomorrow!

File: src/api/app.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (janjaali) [reply src/api/app.js:46]: Yeah, sorry just misread the position of the authentication block, looks good to me :)

File: src/ui/mails/Mails.js (hunk 1 above)
Comment (rdehuyss) [reply src/ui/mails/Mails.js]: Fixed!

**Comment by rdehuyss** [comment]
You can now also search using subject and pass them as parameters in the UI to directly filter on these values.

File: src/ui/mails/Mails.js (hunk 2 above)
Comment (janjaali) [reply src/ui/mails/Mails.js]: 👍

File: src/ui/mails/Mails.js (hunk 1 above)
Comment (janjaali) [reply src/ui/mails/Mails.js]: 👍

**Review by janjaali — APPROVED** [review]

**Comment by janjaali** [comment]
Thanks for contributing, I will draft a new release containing all this newly added features ;)

**Comment by janjaali** [comment]
Pushed a new release <https://github.com/janjaali/sendGrid-mock/releases/tag/v1.2.0>. Happy hacking ;)

**Pull request merged**